### PR TITLE
Update Authentication_Cheat_Sheet.md

### PR DESCRIPTION
**Pull request by aiacobelli2**
I've tried to add/correct the first 4 titles.

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [x] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [x] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [x] All your assets are stored in the **assets** folder.
- [x] All the images used are in the **PNG** format.
- [x] Any references to websites have been formatted as `[TEXT](URL)`
- [x] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

**Comment by aiacobelli2** [comment]
> I'm a little concerned that this is too jargony. Cheat Sheets are meant to be easy to read and understand and I'm afraid this will be too hard for beginners to parse.

I understand, but I've seen lot's of Appsec beginners not understanding the basic difference between identification, authentication, digital identity, and identity proofing. This basic misconception makes them not understand how to propperly think about the subject. 

The same about user-ids and usernames.

**Comment by szh** [comment]
I do understand the concern, but I think we need to have a way of explaining the difference in a way that beginners can understand. This may be a matter of using simple language and the like.

File: cheatsheets/Authentication_Cheat_Sheet.md [hunk 1]
```diff
@@ -2,23 +2,31 @@
 
 ## Introduction
 
-**Authentication** is the process of verifying that an individual, entity, or website is who/what it claims to be. Authentication in the context of web applications is commonly performed by submitting a username or ID and one or more items of private information that only a given user should know.
+**Authentication** is the process of determining the validity of one or more authenticators used to claim a digital identity. Authentication establishes that a subject attempting to access a digital service is in control of the technologies used to authenticate. 
+
+**Digital Identity** is the unique representation of a subject engaged in an online transaction. A digital identity is always unique in the context of a digital service but does not necessarily need to be traceable back to a specific real-life subject.
+
+**Identity Proofing** establishes that a subject is actually who they claim to be. This concept is related to KYC concepts and it aims to bind a digital identity with a real person.
```
Comment (otkd) [inline cheatsheets/Authentication_Cheat_Sheet.md:9]:
```suggestion
```

Too detailed for audience

File: cheatsheets/Authentication_Cheat_Sheet.md [hunk 2]
```diff
@@ -2,23 +2,31 @@
 
 ## Introduction
 
-**Authentication** is the process of verifying that an individual, entity, or website is who/what it claims to be. Authentication in the context of web applications is commonly performed by submitting a username or ID and one or more items of private information that only a given user should know.
+**Authentication** is the process of determining the validity of one or more authenticators used to claim a digital identity. Authentication establishes that a subject attempting to access a digital service is in control of the technologies used to authenticate.
```
Comment (otkd) [inline cheatsheets/Authentication_Cheat_Sheet.md:5]:
```suggestion
**Authentication** is the process of verifying that an individual, entity, or website is who or what it claims to be by determining the validity of one or more authenticators (like passwords, fingerprints, or security tokens) that are used to back up this claim.
```

**Review by otkd — CHANGES_REQUESTED** [review]
@aiacobelli2 Agree conceptually with your changes, but simplified for intended audience as @szh noted

**Comment by szh** [comment]
@otkd I like your suggestions. @aiacobelli2 would you like to incorporate them and update the PR?

**Comment by jmanico** [comment]
> > I'm a little concerned that this is too jargony. Cheat Sheets are meant to be easy to read and understand and I'm afraid this will be too hard for beginners to parse.
> 
> I understand, but I've seen lot's of Appsec beginners not understanding the basic difference between identification, authentication, digital identity, and identity proofing. This basic misconception makes them not understand how to properly think about the subject. The same about user id's and usernames.
> 
> For example, in this cheat-sheet appears Oauth as an authentication framework and this is just wrong.

Your comment about OAuth is spot on, thank you for catching this!

**Comment by jmanico** [comment]
These changes are super important. We have big mistakes regarding how we define these core terms and I totally support this work. Thank you everyone!

File: cheatsheets/Authentication_Cheat_Sheet.md (hunk 2 above)
Comment (otkd) [reply cheatsheets/Authentication_Cheat_Sheet.md:5]: Updated suggestion to include `(**AuthN**)`

**Review by jmanico — APPROVED** [review]

**Comment by jmanico** [comment]
There are a few lint errors in the markup. Can you kindly fix those? I'll merge this soon after they are fixed.

**Review by szh — CHANGES_REQUESTED** [review]
@jmanico I think we should wait to merge this until @otkd's suggestions are implemented.

**Comment by jmanico** [comment]
> @jmanico I think we should wait to merge this until @otkd's suggestions are implemented.

You got it, I have a quick trigger-finger to merge, but I will wait! :)

**Review by jmanico — APPROVED** [review]

**Comment by jmanico** [comment]
There are a few lint errors here but we are getting close!

File: cheatsheets/Authentication_Cheat_Sheet.md [hunk 3]
```diff
@@ -14,15 +14,13 @@
...
+The primary function of a User ID is to uniquely identify a user within a system. Ideally, User IDs should be randomly generated to prevent the creation of predictable or sequential IDs, which could pose a security risk, especially in systems where User IDs might be exposed or inferred from external sources.
 
-Ideally, IDs should be unique and, if possible, ungessable. This is why using a PRNG (not necessarily a CSPRNG) is usually a good idea. UUID4 is a fine algorithm to use for this purpose. The trade-off of this decision is usually performance in terms of user creation and user search. 
+### Usernames
 
-### UserNames
+Usernames are easy-to-remember identifiers chosen by the user and used for identifying themselves when logging into a system or service. The terms User ID and username might be used interchangeably if the username chosen by the user also serves as their unique identifier within the system.
```
Comment (otkd) [inline cheatsheets/Authentication_Cheat_Sheet.md:21]:
```suggestion
Usernames are easy-to-remember identifiers chosen by the user and used for identifying themselves when logging into a system or service. The terms User ID and username might be used interchangeably if the username chosen by the user also serves as their unique identifier within the system.
```

Should resolve mdlint check

**Review by otkd — CHANGES_REQUESTED** [review]

**Pull request merged**